lib/src/model.dart

```diff
@@ -2729,16 +2729,42 @@ abstract class ModelElement extends Canonicalization
 
   /// Returns linked annotations from a given metadata set, with escaping.
   List<String> annotationsFromMetadata(List<ElementAnnotation> md) {
-    if (md == null) return <String>[];
-    return md.map((ElementAnnotation a) {
+    List<String> annotationStrings = [];
+    if (md == null) return annotationStrings;
+    for (ElementAnnotation a in md) {
       String annotation = (const HtmlEscape()).convert(a.toSource());
-      // a.element can be null if the element can't be resolved.
-      var me = packageGraph
-          .findCanonicalModelElementFor(a.element?.enclosingElement);
-      if (me != null)
-        annotation = annotation.replaceFirst(me.name, me.linkedName);
-      return annotation;
-    }).toList(growable: false);
+      Element annotationElement = a.element;
+
+      ClassElement annotationClassElement;
+      if (annotationElement is ExecutableElement) {
+        annotationElement = (annotationElement as ExecutableElement)
```

Comment:
I'm surprised that type promotion doesn't just allow you to type `annotationElement` w/o the cast.

Reply:
It might be because of the assignment of annotationElement to a different type, but yes it doesn't promote the type here.

Reply:
It doesn't promote because you are assigning to the variable inside the guarded block.